docs/architecture/component_templates.md

```diff
@@ -0,0 +1,134 @@
+
+An account component template can have multiple storage entries. A storage entry can specify either a **single-slot value**, a **multi-slot value**, or a **storage map**.
+
+Each of these storage entries contain the following fields:
+
+- `name`: A name for identifying the storage entry
+- `description` (optional): Describes the intended function of the storage slot within the component definition
+
+Additionally, based on the type of the storage entry, there are specific fields that should be specified.
+
+##### Single-slot value
+
+A single-slot value fits within one slot (i.e., one word).
+
+For a single-slot entry, the following fields are expected:
+
+- `slot`: Specifies the slot index in which the value will be placed
+- `value` (optional): Contains the initial storage value for this slot. Will be interpreted as a `word` unless another `type` is specified
+- `type` (optional): Describes the expected type for the slot
+
+If no `value` is provided, the entry acts as a placeholder, requiring a value to be passed at instantiation. In this case, specifying a `type` is mandatory to ensure the input is correctly parsed.
+Valid types for a single-slot value are `word` or `auth::rpo_falcon512::pub_key`.
+
+Note that at least one of `value` and `type` should be specified. 
+
+In the above example, the first and second storage entries are single-slot values.
+
+##### Multi-slot value
+
+Multi-slot values are composite values that exceed the size of a single slot (i.e., more than one `word`).
+
+For multi-slot values, the following fields are expected:
+
+- `slots`: Specifies the list of slots that the value comprises
+- `values`: Contains the initial storage value for the specified slots
+
+Placeholders can currently not be defined for multi-slot values.
+
+##### Storage maps
+
+Storage maps consist of key-value pairs, where both keys and values are single words.
+
+Storage map entries can specify the following fields:
+
+- `slot`: Specifies the slot index in which the value will be placed
+- `values`: Contains a list of map entries, defined by a `key` and `value `
+
+Storage maps `key`s and `value`s are word-long values, which can be defined as placeholders.
+
+In the example, the third storage entry defines a storage map.
+
+#### Specifying values and their types
+
+In the TOML format, any value that is one word long can be written as a single value, or as exactly four field elements. In turn, a field element is a number within Miden's finite field. 
+
+A word can be written as a hexadecimal value, and field elements can be written either as hexadecimal or decimal numbers.
+
+In our example, the `token_metadata` single-slot entry is defined as four elements, where the first element is a placeholder, and the second, third and fourth are hardcoded values.
+
+##### Word-long types
+
+Valid word-long types are `word` (default type) and `auth::rpo_falcon512::pub_key` (represents a Falcon public key). Both can be parsed as hexadecimal strings.
```

Comment:
```suggestion
Valid word types are `word` (default type) and `auth::rpo_falcon512::pub_key` (represents a Falcon public key). Both can be written as hexadecimal strings.
```

Do we need "word-long" or is "word" clear enough?

Nit: Users probably care about what they can write rather than what our code can parse.

Reply:
I wanted to differentiate between the `word` template type, and word as a general VM concept (also applies to your other comment about how it's being stylized). There will also be types that are more than one word long. But it's a good point and I'm not 100% sure about the wording here, so if you think it's better as you are suggesting I can change it.